lasagne/layers/recurrent.py

```diff
@@ -230,18 +282,19 @@ def __init__(self, incoming, input_to_hidden, hidden_to_hidden,
         # Get the input dimensionality and number of units based on the
         # expected output of the input-to-hidden layer
         self.num_inputs = np.prod(input_shape[2:])
```

Comment:
That will break if `input_shape[2:]` contains `None`s. Is `self.num_inputs` needed anywhere?